WordPress/AbstractThemeSniff.php

```diff
@@ -0,0 +1,1068 @@
+<?php
+/**
+ * Represents a PHP_CodeSniffer sniff for sniffing WordPress coding standards.
+ *
+ * @category  PHP
+ * @package   PHP_CodeSniffer
+ */
+
+/**
+ * Represents a PHP_CodeSniffer sniff for sniffing WordPress coding standards.
+ *
+ * Provides a bootstrap for the sniffs, to reduce code duplication.
+ *
+ * @category  PHP
+ * @package   PHP_CodeSniffer
+ * @author    Juliette Reinders Folmer <wpplugins_nospam@adviesenzo.nl>
+ * @author    Kevin Haig
+ * @author    Ulrich Pogson <ulrich@pogson.ch>
+ */
+abstract class WordPress_AbstractThemeSniff implements PHP_CodeSniffer_Sniff {
+
+	/**
+	 * Array holding the once through WordPress Theme file checks.
+	 *
+	 * @var array
+	 */
+	private static $sniff_helper;
+
+	/**
+	 * Initial setup.
+	 *
+	 * @var array $theme_data contains data from style.css headers.
+	 */
+	public function __construct() {
+		// do a once through pass and get the files.
+		if ( empty( self::$sniff_helper ) || ! is_array( self::$sniff_helper ) || false === self::$sniff_helper ) {
+			$files = $this->get_files();
+			$themedir = $files[0];
+			// Only run this if checking a directory.
+			$themedir_parts = pathinfo( $themedir );
+			if ( isset( $themedir_parts['extension'] ) ) {
+				return;
```

Comment:
You are in a constructor, so returning does not work. You would need to throw an exception.